hugo/layouts/_default/list.algolia.json

```diff
@@ -0,0 +1,53 @@
+{{/* Generates a valid Algolia search index */}}
+{{- $section := $.Site.GetPage "section" .Section }}
+[
+    {{ range $idx, $page := $.Site.Pages -}}
+    {{- if $idx -}},{{- end -}}
+        {{- if or (and ($page.IsDescendant $section) (and (not $page.Draft) (not $page.Params.private))) $section.IsHome -}}
+            {{- template "algolia-object" (dict
+                "index" 0
+                "page" $page
+            ) -}}
+        {{- end -}}
+    {{- end -}}
+]
+
+{{- define "algolia-object" -}}
+    {{ $page := .page -}}
+    {{ $cutoff := 1000 -}}
+    {{ $index := .index -}}
+    {{ $skipWords := mul $cutoff $index }}
+
+    {{- $objectID := (print $page.File.UniqueID "_" (add $index 1)) -}}
+    {{- $title :=  $page.Title -}}
+    {{- $link := $page.Permalink -}}
+    {{- $categories := cond (isset $page.Params "categories") ($page.Params.categories) "" -}}
+    {{- $order := (add $index 1) -}}
+    {{- $publishDate := $page.PublishDate -}}
+    {{- $summary := $page.Summary -}}
+    {{- $content := "" -}}
+
+    {{- if eq $index 0 -}}
+        {{- $content = delimit (first $cutoff $page.PlainWords) " " -}}
```

Comment:
The current approach simply appears to simply strip markdown/hugo markup from the input. However this includes code blocks and their output in the index. Is this intentional?

Reply:
Yes we're currently using hugo's plain function. This strips all html and markdown and keeps the raw content including everything inside shortcodes. People might be searching on code pieces? They will also show up. If we don't want this we have to look into building a custom hugo function but I'm not sure that's possible in hugo. (https://discourse.gohugo.io/t/adding-custom-functions/14164)